### PR TITLE
Keep egress info for 24h

### PR DESCRIPTION
**Pull request by frostbyte73**

### CODE REVIEW

File: pkg/service/egress.go [hunk 1]
```diff
@@ -41,10 +42,12 @@ func NewEgressService(
 	}
 }
 
-func (s *EgressService) Start() {
-	if s.rpcClient != nil {
-		go s.updateWorker()
+func (s *EgressService) Start() error {
+	if s.rpcClient != nil && s.es != nil {
```
Comment (davidzhao) [inline pkg/service/egress.go:46]: is there a way to prevent this from being called multiple times?

File: pkg/service/egress.go [hunk 2]
```diff
@@ -236,58 +223,88 @@ func (s *EgressService) StopEgress(ctx context.Context, req *livekit.StopEgressR
...
+					logger.Errorw("failed to read results", err)
+					continue
 				}
 
-				// log results
-				if res.Error != "" {
-					logger.Errorw("egress failed", errors.New(res.Error), "egressID", res.EgressId)
-				} else {
-					logger.Infow("egress ended", "egressID", res.EgressId)
+				ensureRoomName(res)
```
Comment (davidzhao) [inline pkg/service/egress.go:252]: is this for backwards compatibility?

File: pkg/service/redisstore.go [hunk 3]
```diff
@@ -227,17 +256,58 @@ func (s *RedisStore) DeleteParticipant(_ context.Context, roomName livekit.RoomN
 	return s.rc.HDel(s.ctx, key, string(identity)).Err()
 }
 
+// Temporary migration for upgrading from LiveKit v1.1.2 to v1.1.3
+func (s *RedisStore) MigrateEgressInfo() (int, error) {
```
Comment (davidzhao) [inline pkg/service/redisstore.go:260]: do we have any sort of marker (maybe in redis) indicating the last version of livekit? might be useful for future "migration" efforts.

Comment (davidzhao) [reply pkg/service/redisstore.go:260]: also. wondering how this works when multiple instances of lk server are running at the same time.

File: pkg/service/redisstore_test.go [hunk 4]
```diff
@@ -111,7 +113,101 @@ func TestRoomLock(t *testing.T) {
 	})
 }
 
-func TestStoreIngress(t *testing.T) {
+func TestEgressStore(t *testing.T) {
```
Comment (davidzhao) [inline pkg/service/redisstore_test.go:116]: YES!

**Review by davidzhao — APPROVED** [review]

File: pkg/service/egress.go (hunk 2 above)
Comment (frostbyte73) [reply pkg/service/egress.go:252]: yeah, left a comment above the new function

File: pkg/service/redisstore.go (hunk 3 above)
Comment (frostbyte73) [reply pkg/service/redisstore.go:274]: added version and migration locking

**Review by davidzhao — APPROVED** [review]
lgtm!

**Pull request merged**